frontend/scripts/componentSchemas/schemaUtils.ts

```diff
@@ -1,4 +1,57 @@
 import jsonpointer from 'jsonpointer';
+import { sortTextResourceBindings } from './languageUtils';
+
+export const allPropertyKeys = [];
+
+export const generateComponentSchema = (
+  componentName: string,
+  layoutSchema: any,
+  version: string,
+) => {
+  const definitionName = `Comp${componentName}`;
+  const componentSchema = getComponentSchema(definitionName, layoutSchema, version);
+
+  let schema: any = {
+    $id: `https://altinncdn.no/schemas/json/component/${componentName}.schema.v1.json`,
+    $schema: layoutSchema.$schema,
+  };
+
+  if (componentSchema.allOf) {
+    schema = { ...schema, ...expandAllOf(componentSchema, layoutSchema) };
+    const expectedProperties = Object.keys(
+      componentSchema.allOf[componentSchema.allOf.length - 1].properties,
+    );
+    allPropertyKeys.push(...expectedProperties);
+
+    if (!verifySchema(schema, expectedProperties)) {
+      return null;
+    }
+  } else if (componentSchema.anyOf) {
+    schema.anyOf = expandAnyOf(componentSchema, layoutSchema);
+  }
+
+  // Expand all refs in properties
+  schema.properties = expandRefsInProperties(schema.properties, layoutSchema);
+
+  // Sort text resource binding keys
+  if (schema.properties?.textResourceBindings) {
+    schema.properties.textResourceBindings.properties = sortTextResourceBindings(
+      schema.properties.textResourceBindings.properties,
+    );
+  }
+
+  schema.title = `${componentName} component schema`;
+  return schema;
+};
```

Comment:
This function does a lot of different things, can we maybe split it into smaller chunks (one for each thing that is being done)? 😄 

Something like this. Maybe it can be done even more? 🤔 
```suggestion
export const generateComponentSchema = (
  componentName: string,
  layoutSchema: SomeType, // Should replace this with a type
  version: string,
) => {
  const definitionName: string = `Comp${componentName}`;
  const componentSchema: SomeType = getComponentSchema(definitionName, layoutSchema, version);

  let schema: SomeType = getSchema(componentName, layoutSchema);
 
  if (componentSchema.allOf) {
    schema = { ...schema, ...expandAllOf(componentSchema, layoutSchema) };
    const expectedProperties: SomeType = getExpectedProperties(componentSchema);
    allPropertyKeys.push(...expectedProperties);

    if (!verifySchema(schema, expectedProperties)) {
      return null;
    }
  } else if (componentSchema.anyOf) {
    schema.anyOf = expandAnyOf(componentSchema, layoutSchema);
  }

  // Expand all refs in properties
  schema.properties = expandRefsInProperties(schema.properties, layoutSchema);

  // Sort text resource binding keys
  if (schema.properties?.textResourceBindings) {
    schema.properties.textResourceBindings.properties = sortTextResourceBindings(
      schema.properties.textResourceBindings.properties,
    );
  }

  schema.title = `${componentName} component schema`;
  return schema;
};

const getSchema = (componentName: string, layoutSchema: SomeType): SomeType => ({
    $id: `https://altinncdn.no/schemas/json/component/${componentName}.schema.v1.json`,
    $schema: layoutSchema.$schema,
});

const getExpectedProperties = (componentSchema: SomeType): SomeType => {
    return Object.keys(
      componentSchema.allOf[componentSchema.allOf.length - 1].properties,
    );
}
```

Reply:
Great job @ErlingHauan! 👏 This looks a lot cleaner!